src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+    elif unit == "ms" or unit == "milliseconds":
+        return "MilliSecond"
+    elif unit == "us{CPU}":
+        return "MicroSecond"
+    elif unit == "10^2.%":
+        return "Percent"
+    elif unit == "%":
+        return "Percent"
+    elif unit == "percent":
+        return "Percent"
+    elif unit == "1/s":
+        return "PerSecond"
+    elif unit == "frames/seconds":
+        return "PerSecond"
+    elif unit == "s{CPU}":
+        return "Second"
+    elif unit == "s{uptime}":
+        return "Second"
+    elif unit == "{dBm}":
+        return "DecibelMilliWatt"
+    else:
+        return "Unspecified"
+    
+
+def type_metrickind_caster(metrickind: str, valuetype: str) -> str:
+    if metrickind == "GAUGE":
+        return "gauge"
+    if metrickind == "DELTA" and valuetype == "DISTRIBUTION":
+        return "gauge"
+    if metrickind == "DELTA" and valuetype != "DISTRIBUTION":
+        return "count,delta"
+    else:
+        return ""
+
+    
+def metric_name_parse(metric_name: str,metric_type: str) -> str:
+   # . -> _
+   # / -> .
+   metric_name.replace(r'\.','_')
+   metric_name.replace(r'/','.')
+   return metric_name
+
+async def init_session():
+    async with init_gcp_client_session() as gcp_session:
+        token = await create_token(logging_context, gcp_session)
+        if token:
+            await get_metric_descriptors(gcp_session,token)
+        else:
+            logging_context.log('Unable to acquire authorization token.')
+
+
+async def _sent_metadata_to_dt(dynatrace_url: Text,dynatrace_access_key, dt_session, metric_metadata_list):
+    data_input = "\n".join([metadata.to_json() for metadata in metric_metadata_list])
+
+    dt_url = f"{dynatrace_url.rstrip('/')}/api/v2/settings/objects"
+    response = await dt_session.post(
+        url=dt_url,
+        headers={
+            "Authorization": f"Api-Token {dynatrace_access_key}",
+            "Content-Type": "application/json"
+        },
+        data=data_input,
+        verify_ssl=get_should_require_valid_certificate()
+    )
+
+    if response.status != 200:
+        print("Error During sending metadata meterics")
+        return False
+    else:
+        return True
+
+
+async def get_metric_descriptors(gcp_session,token):
+    headers = {
+    "Accept": "application/json",
+    "Authorization": f"Bearer {token}"}
+    url = f"https://monitoring.googleapis.com/v3/projects/{projectID}/metricDescriptors"
+    resp = await gcp_session.request('GET', url=url, headers=headers)
+    page = await resp.json()
+
+    metric_yaml = [
+        {
+            "value": x.get("type"),
+            "key":  "cloud.gcp." + metric_name_parse(x.get("type"),type_metrickind_caster(x.get("metricKind"),x.get("valueType"))),
+            "displayName":  x.get("displayName"),
+            "name": x.get("displayName"),
+            "description": x.get("description"),
+            "type": type_metrickind_caster(x.get("metricKind"),x.get("valueType")),
```

Comment:
Can we have method more descriptive? It is casting from GCP kind to DT kind, right? Can you put that in the name?